### PR TITLE
ref(server): Rate limit metrics in project cache

### PR DESCRIPTION
**Pull request by jan-auer**
Moves rate limiting of metric buckets from cached rate limits and static quotas
into the project cache. This avoids messages to the processor in the case that a
project is rate limited.

#skip-changelog

### CODE REVIEW

File: relay-server/src/actors/processor.rs [hunk 1]
```diff
@@ -1356,88 +1351,62 @@ impl EnvelopeProcessorService {
...
-            // Check with redis if the throughput limit has been exceeded, while also updating
-            // the count so that other relays will be updated too.
-            match rate_limiter.is_rate_limited(quotas, item_scoping, total_buckets, false) {
-                Ok(limits) if limits.is_limited() => {
-                    relay_log::debug!(
-                        "dropping {total_buckets} buckets due to throughput ratelimit"
-                    );
+        // Check with redis if the throughput limit has been exceeded, while also updating
+        // the count so that other relays will be updated too.
+        match rate_limiter.is_rate_limited(quotas, item_scoping, quantities.buckets, false) {
```
Comment (jan-auer) [inline relay-server/src/actors/processor.rs:1381]: Note to reviewers: Look at this without whitespace diffs.

File: relay-server/src/actors/processor.rs [hunk 2]
```diff
@@ -1272,7 +1267,7 @@ impl EnvelopeProcessorService {
         let scoping = *bucket_limiter.scoping();
 
         if let Some(rate_limiter) = self.inner.rate_limiter.as_ref() {
-            let item_scoping = ItemScoping {
+            let item_scoping = relay_quotas::ItemScoping {
```
Comment (jjbayer) [inline relay-server/src/actors/processor.rs:1270]: Why is `Scoping` imported, but `ItemScoping` isn't?

**Review by jjbayer — APPROVED** [review]
Nice!

File: relay-server/src/actors/processor.rs (hunk 2 above)
Comment (jan-auer) [reply relay-server/src/actors/processor.rs:1270]: It's used exclusively with `#[cfg(feature = "processing")]`, so the local qualifier was the simplest fix. We could also do a conditional import or a local import

Comment (jjbayer) [reply relay-server/src/actors/processor.rs:1270]: Ah, right.

**Pull request merged**